include/nccl_ofi_config_bottom.h

```diff
@@ -9,7 +9,11 @@
 #define OFI_LIKELY(x)   __builtin_expect((x), 1)
 #define OFI_UNLIKELY(x) __builtin_expect((x), 0)
 
+#ifdef __cplusplus
+#define NCCL_OFI_EXPORT_SYMBOL __attribute__((visibility("default"))) [[maybe_unused]]
```

Comment:
My comment from https://github.com/aws/aws-ofi-nccl/pull/521/commits/19d4b3847396957b3a6d7ae9668a80180ae587fe remains; this shouldn't be necessary and I'd like to get to root cause before committing it.